pkg/gitreceive/git/sha.go

```diff
@@ -0,0 +1,31 @@
+package git
+
+import (
+	"fmt"
+)
+
+const (
+	// this constant represents the length of a shortened git sha - 8 characters long
+	shortShaIdx = 8
+	fullShaLen  = 40
+)
+
+type ErrGitShaTooShort struct {
+	sha string
+}
+
+func (e ErrGitShaTooShort) Error() string {
+	return fmt.Sprintf("git sha %s was too short", e.sha)
+}
+
+type SHA struct {
+	Full  string
+	Short string
+}
```

Comment:
I'm probably overthinking this, but assuming a `SHA` is meant to be immutable, should these fields maybe no be exported and have exported accessors instead?